cirq/google/calibration/workflow.py

```diff
@@ -11,21 +11,210 @@
+    """Describes a given moment in terms of a Floquet characterization request.
+
+    Args:
+        moment: Moment to characterize.
+        options: Options that are applied to each characterized gate within a moment.
+        gates_translator: Function that translates a gate to a supported FSimGate which will undergo
+            characterization. Defaults to sqrt_iswap_gates_translator.
+        pairs_in_canonical_order: Whether to sort each of the qubit pair so that the first qubit
+            is always lower than the second.
+        pairs_sorted: Whether to sort all the qutibt pairs extracted from the moment which will
+            undergo characterization.
+
+    Returns:
+        Instance of FloquetPhasedFSimCalibrationRequest that characterizes a given moment, or None
+        when it is an empty, measurement or single-qubit gates only moment.
+
+    Raises:
+        IncompatibleMomentError when a moment contains operations other than the operations matched
+        by gates_translator, or it mixes a single qubit and two qubit gates.
+    """
+
+    measurement = False
+    single_qubit = False
+    gate: Optional[FSimGate] = None
+    pairs = []
+
+    for op in moment:
+        if not isinstance(op, GateOperation):
+            raise IncompatibleMomentError('Moment contains operation different than GateOperation')
+
+        if isinstance(op.gate, MeasurementGate):
+            measurement = True
+        elif isinstance(op.gate, SingleQubitGate):
+            single_qubit = True
+        else:
+            translated_gate = gates_translator(op.gate)
+            if translated_gate is None:
+                raise IncompatibleMomentError(
+                    f'Moment {moment} contains unsupported non-single qubit operation {op}'
+                )
+            elif gate is not None and gate != translated_gate:
+                raise IncompatibleMomentError(
+                    f'Moment {moment} contains operations resolved to two different gates {gate} '
+                    f'and {translated_gate}'
+                )
+            else:
+                gate = translated_gate
+            pair = cast(
+                Tuple[Qid, Qid], tuple(sorted(op.qubits) if pairs_in_canonical_order else op.qubits)
+            )
+            pairs.append(pair)
+
+    if gate is None:
+        # Either empty, single-qubit or measurement moment.
+        return None
+
+    if gate is not None and (measurement or single_qubit):
+        raise IncompatibleMomentError(
+            f'Moment contains mixed two-qubit operations and '
+            f'single-qubit operations or measurement operations.'
+        )
+
+    return FloquetPhasedFSimCalibrationRequest(
+        pairs=tuple(sorted(pairs) if pairs_sorted else pairs), gate=gate, options=options
+    )
+
+
+def floquet_characterization_for_circuit(
+    circuit: Circuit,
+    options: FloquetPhasedFSimCalibrationOptions = FloquetPhasedFSimCalibrationOptions.all_except_for_chi_options(),
```

Comment:
weird to have a function call in the default argument. Maybe instead of static functions, we can have those two Options be global variables, e.g. `ALL_EXCEPT_CHI_OPTIONS`

Reply:
Done.